apps/endatix-hub/features/public-form/ui/survey-component.tsx

```diff
@@ -58,7 +60,57 @@ export default function SurveyComponent({ definition, formId, submission }: Surv
     });
   }, []);
 
+  const interceptImages = useCallback((sender: SurveyModel, options: ValueChangingEvent) => {
+    const resizeImage = (imgSrc: string, type: string): Promise<string> => {
+      return new Promise((resolve) => {
+        const img = new Image();
+        img.onload = () => {
+          let width = img.width;
+          let height = img.height;
+    
+          if (width > height) {
+            if (width > MAX_IMAGE_SIZE) {
+              height = height * (MAX_IMAGE_SIZE / width);
+              width = MAX_IMAGE_SIZE;
+            }
+          } else {
+            if (height > MAX_IMAGE_SIZE) {
+              width = width * (MAX_IMAGE_SIZE / height);
+              height = MAX_IMAGE_SIZE;
+            }
+          }
+    
+          let canvas = document.createElement("canvas");
+          canvas.width = width;
+          canvas.height = height;
+          const ctx = canvas.getContext("2d");
+    
+          if (ctx) {
+            ctx.drawImage(img, 0, 0, width, height);
+            resolve(canvas.toDataURL(type));
+          }
+        };
+        img.src = imgSrc;
+      });
+    };
+    
+    const processImages = async () => {
+    
+      for (let i = 0; i < options.question.value.length; i++) {
+        if (options.value[i].type.includes('image')) {
+            options.value[i].content = await resizeImage(
+              options.value[i].content,
```

Comment:
Having a local variable will be faster and easier to read instead of going through the array reference every time. Can you refacto to extract a variable for `options.value[i]`